node/full_client.go

```diff
@@ -412,12 +412,23 @@ func (c *FullClient) Health(ctx context.Context) (*ctypes.ResultHealth, error) {
 // Block method returns BlockID and block itself for given height.
 //
 // If height is nil, it returns information about last known block.
+// If height is a valid block tag, it returns information about the tagged
+// block, see BlockNumber for details.
 func (c *FullClient) Block(ctx context.Context, height *int64) (*ctypes.ResultBlock, error) {
-	heightValue := c.normalizeHeight(height)
+	var heightValue uint64
+
+	switch {
+	// block tag = included
+	case height != nil && *height == -1:
+		heightValue = c.node.blockManager.GetDAIncludedHeight(ctx)
+	default:
+		heightValue = c.normalizeHeight(height)
+	}
```

Comment:
Imo we are over complicating what the `height` argument is. 

`Block` should just grab the rollup block by rollup height, or grab the latest rollup block if no height is provided. 

I think it would be cleaner to have a separate method that is called when a user is specifically checking for a DA included block.